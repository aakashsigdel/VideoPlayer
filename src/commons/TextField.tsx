import * as React from 'react'

import styles from './TextField.module.css'

export interface ITextFieldProps {
  id: string
  label: string
  type: 'text' | 'url'
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
}
export const TextField: React.FunctionComponent<ITextFieldProps> = ({
  id,
  label,
  type,
  onChange,
}) => (
  <div className={styles.inputContainer}>
    <label htmlFor={id}>{label}</label>
    <input
      type={type}
      id={id}
      onChange={onChange}
      required
      className={styles.input}
    />
  </div>
)
