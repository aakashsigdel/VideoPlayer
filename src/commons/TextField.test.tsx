import * as React from 'react'
import { create } from 'react-test-renderer'

import { TextField, ITextFieldProps } from './TextField'

describe('<TextField />', () => {
  const defaultProps: ITextFieldProps = {
    id: 'test',
    label: 'testLabel',
    type: 'text',
    onChange: jest.fn(),
  }

  it('should render correctly', () => {
    const tree = create(<TextField {...defaultProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
