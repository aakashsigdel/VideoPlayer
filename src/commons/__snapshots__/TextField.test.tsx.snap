// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render correctly 1`] = `
<div
  className="inputContainer"
>
  <label
    htmlFor="test"
  >
    testLabel
  </label>
  <input
    className="input"
    id="test"
    onChange={[MockFunction]}
    required={true}
    type="text"
  />
</div>
`;
